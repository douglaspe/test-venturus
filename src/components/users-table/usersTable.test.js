import React from 'react'
import { shallow } from 'enzyme';

import UsersTable from './usersTable';
import UsersList from '../users-list/usersList';

describe('<UsersTable />', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<UsersTable />));

    it('should render the tag <UsersTable />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render the <UsersList /> Component', () => {
        expect(wrapper.containsMatchingElement(<UsersList />)).toEqual(true);
    });


}); 
