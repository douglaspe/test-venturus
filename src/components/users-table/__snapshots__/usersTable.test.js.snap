// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="table table-custom"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              UserName
            </th>
            <th
              scope="col"
            >
              Name
            </th>
            <th
              scope="col"
            >
              E-mail
            </th>
            <th
              scope="col"
            >
              City
            </th>
          </tr>
        </thead>
        <tbody>
          <UsersList />
        </tbody>
      </table>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th
                scope="col"
              >
                UserName
              </th>
              <th
                scope="col"
              >
                Name
              </th>
              <th
                scope="col"
              >
                E-mail
              </th>
              <th
                scope="col"
              >
                City
              </th>
            </tr>
          </thead>,
          <tbody>
            <UsersList />
          </tbody>,
        ],
        "className": "table table-custom",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th
                scope="col"
              >
                UserName
              </th>
              <th
                scope="col"
              >
                Name
              </th>
              <th
                scope="col"
              >
                E-mail
              </th>
              <th
                scope="col"
              >
                City
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="col"
                >
                  UserName
                </th>,
                <th
                  scope="col"
                >
                  Name
                </th>,
                <th
                  scope="col"
                >
                  E-mail
                </th>,
                <th
                  scope="col"
                >
                  City
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "UserName",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "UserName",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "E-mail",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "E-mail",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "City",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "City",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <UsersList />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="table table-custom"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                UserName
              </th>
              <th
                scope="col"
              >
                Name
              </th>
              <th
                scope="col"
              >
                E-mail
              </th>
              <th
                scope="col"
              >
                City
              </th>
            </tr>
          </thead>
          <tbody>
            <UsersList />
          </tbody>
        </table>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  UserName
                </th>
                <th
                  scope="col"
                >
                  Name
                </th>
                <th
                  scope="col"
                >
                  E-mail
                </th>
                <th
                  scope="col"
                >
                  City
                </th>
              </tr>
            </thead>,
            <tbody>
              <UsersList />
            </tbody>,
          ],
          "className": "table table-custom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                  scope="col"
                >
                  UserName
                </th>
                <th
                  scope="col"
                >
                  Name
                </th>
                <th
                  scope="col"
                >
                  E-mail
                </th>
                <th
                  scope="col"
                >
                  City
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    scope="col"
                  >
                    UserName
                  </th>,
                  <th
                    scope="col"
                  >
                    Name
                  </th>,
                  <th
                    scope="col"
                  >
                    E-mail
                  </th>,
                  <th
                    scope="col"
                  >
                    City
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "UserName",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "UserName",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "E-mail",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "E-mail",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "City",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "City",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <UsersList />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
